### DISCUSSION TITLE
Need help with issues with transferring an existing domain to GitHub

### DISCUSSION THREAD

**Discussion by thescribe17** [community/community#140006] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi Team,

I have had an existing website for over a decade and recently decided to transfer it to GitHub. I followed the instructions on the [Managing a Custom Domain for Your GitHub pages site](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site) page. However, as I did not have direct access to make the changes myself, I asked the DNS provider to create the A record to point to the 4 listed IP addresses for GitHub Pages, which they did.

However, the DNS check is unsuccessful and I get the following error message:
_Both karenleefield.com and its alternate name are improperly configured
Domain does not resolve to the GitHub Pages server. For more information, see [documentation](https://docs.github.com/articles/setting-up-a-custom-domain-with-github-pages/) (NotServedByPagesError)._

I read on another page that CNAME should point the domain name to my GitHub user name. I contacted the DNS provider again and asked if they could add that too and I got the following response:
_Regarding in updating your A records our technical team has already tried to update it, unfortunately while it is under propagation the hosting service has been cancelled._

I don't understand this, but they have given me an EPP code, yet I can't find anything in the documents about how to use the code to claim my domain and transfer it here. 

I'm nervous that I've lost control of my domain. I don't know what to do. I have searched the discussions and website but cannot find anything to help me.

Can anyone help me resolve this issue please?

kind regards,
Karen

**Comment by ferid333** [comment]
Hi @thescribe17 

Here are some   steps  you can try to resolve the problem:

1. **A Records and CNAME**: You need to have both A records and a CNAME record properly configured. The A records should point to GitHub’s four IP addresses, and the CNAME should point to your GitHub Pages URL (e.g., `<your-username>.github.io`).

2. **Check Domain Propagation**: DNS changes can take some time to propagate, sometimes up to 48 hours. Use a tool like [DNS Checker](https://dnschecker.org) to verify if the DNS records are properly set.

3. **EPP Code**: The EPP code is typically used to transfer domain ownership to another provider. If your hosting was canceled but you have an EPP code, you might need to transfer your domain to a new registrar. You can do this by following the domain transfer process on a new registrar's website (like Namecheap, GoDaddy, etc.).

4. **GitHub Pages Settings**: Make sure that in your GitHub repository settings (under Pages), your custom domain is correctly added.

If your DNS provider canceled your hosting and gave you an EPP code, it sounds like they’re no longer managing your domain. Transferring your domain to a new registrar might help you regain control and make the necessary DNS changes.

**Comment by thescribe17** [comment]
Someone did respond to my question and gave me a hint for a way forward. Because of that response, I resolved the problem. Thank you.

**Comment by NooruddinLakhani** [comment]
You can use the EPP code to transfer your domain to a registrar that allows DNS management. Then, update both A and CNAME records per GitHub’s instructions. 

I have used [https://register.domains/en/dm/transfer-domains](https://register.domains/en/dm/transfer-domains) which supports the transfer of domains to GitHub Pages, as it enables users to manage the essential DNS records that GitHub Pages requires. For example, GitHub recommends creating specific A and CNAME records to point a custom domain to GitHub's servers. Using Register.domains, you can handle these DNS configurations effectively to direct your domain to GitHub Pages for website hosting. This service would allow you to update or add the A and CNAME records in the DNS settings, aligning them with GitHub's guidelines for custom domains.
